src/lint/collect-header-diagnostics.ts

```diff
@@ -0,0 +1,113 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { getLocation, indexWithinElementToTrueLocation } from './utils';
+
+const ruleId = 'header-format';
+
+export function collectHeaderDiagnostics(
+  dom: any,
+  headers: { element: Element; contents: string }[]
+) {
+  let lintingErrors: LintingError[] = [];
+
+  for (let { element, contents } of headers) {
+    if (!/\(.*\)$/.test(contents) || / Operator \( `[^`]+` \)$/.test(contents)) {
+      continue;
+    }
+
+    let name = contents.substring(0, contents.indexOf('('));
+    let params = contents.substring(contents.indexOf('(') + 1, contents.length - 1);
+
+    if (!/[\S] $/.test(name)) {
+      let { line, column } = indexWithinElementToTrueLocation(
+        getLocation(dom, element),
+        contents,
+        name.length
+      );
+      lintingErrors.push({
+        ruleId,
+        nodeType: element.tagName,
+        line,
+        column,
+        message: 'expected header to have a single space before the argument list',
+      });
+    }
+
+    let nameMatches = [
+      // Runtime Semantics: Foo
+      /^(Runtime|Static) Semantics: [A-Z][A-Za-z0-9/]*\s*$/,
+
+      // Number::foo
+      /^[A-Z][A-Za-z0-9]*::[a-z][A-Za-z0-9]*\s*$/,
+
+      // [[GetOwnProperty]]
+      /^\[\[[A-Z][A-Za-z0-9]*\]\]\s*$/,
+
+      // _NativeError_
+      /^_[A-Z][A-Za-z0-9]*_\s*$/,
+
+      // CreateForInIterator
+      // Object.fromEntries
+      // Array.prototype [ @@iterator ]
```

Comment:
We should support `Array.prototype [ %Symbol.iterator% ]` as well, since we're moving to that form.

Reply:
I'd prefer that we disallow it now and make it legal as part of the PR which changes the format, so we are never in a position where both are accepted.

Reply:
But that's in a different repo. There's a currently open PR. How do you expect that to work?

Reply:
We'd change it here, bump it, release it, and include the change in the upstream PR as part of the process of merging that PR.

Reply:
I don't love it but I'll leave it to @ljharb to fight this fight.